### PR TITLE
Closes #2786: Update phonebook/directory links in global footer.

### LINKED ISSUE #2786: Update global footer with new Directory and Phonebook links
## Motivation
When the Directory is launched on QS2, the phonebook will also launch separately. This determines a need for updated links in the global footer.

## Proposed Resolution
image

### Describe the solution you'd like
1. Change `A-Z Index` link to `Directory` with URL https://directory.arizona.edu
2. Change `Phonebook` URL to https://phonebook.arizona.edu

## Additional context
The externally referenced footer is not updated with changes like these. We need to consider this in the solution as well.
```
<html>
  <body>
    <az-footer>
    <script src="https://cdn.digital.arizona.edu/lib/temp-web-components/1.0.15/az-footer.js"></script>
  </body>
</html>
```

### PR DESCRIPTION
**Pull request by danahertzberg**
## Description
Update global footer with links to directory and phonebook.
Includes DB update.

## Related issues
Closes #2786 

## How to test
1. Enable global footer
2. Ensure "Resources" links include "Directory" and "Phonebook" with correct URLs

Testing on site with global footer already enabled: https://iss-2786-4-dana-qs2.pantheonsite.io/

### Arizona Quickstart (install profile, custom modules, custom theme)
- **Patch release changes**
   - [ ] Bug fix
   - [ ] Accessibility, performance, or security improvement
   - [x] Critical institutional link or brand change
   - [ ] Adding experimental module
   - [ ] Update experimental module
- **Minor release changes**
   - [ ] New feature
   - [ ] Breaking or visual change to existing behavior
   - [ ] Upgrade experimental module to stable
   - [ ] Enable existing module by default or database update
   - [ ] Non-critical brand change
   - [ ] New internal API or API improvement with backwards compatibility
   - [ ] Risky or disruptive cleanup to comply with coding standards
   - [ ] High-risk or disruptive change (requires upgrade path, risks regression, etc.)
- **Other or unknown**
   - [ ] Other or unknown

### Drupal core
- **Patch release changes**
   - [ ] Security update
   - [ ] Patch level release (non-security bug-fix release)
   - [ ] Patch removal that's no longer necessary
- **Minor release changes**
   - [ ] Major or minor level update
- **Other or unknown**
   - [ ] Other or unknown

### Drupal contrib projects
- **Patch release changes**
   - [ ] Security update
   - [ ] Patch or minor level update
   - [ ] Add new module
   - [ ] Patch removal that's no longer necessary
- **Minor release changes**
   - [ ] Major level update
- **Other or unknown**
   - [ ] Other or unknown

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: modules/custom/az_global_footer/az_global_footer.install [hunk 1]
```diff
@@ -77,3 +77,71 @@ function az_global_footer_update_1020701() {
...
+
+  foreach ($menu_link_content_ids as $id) {
+    /**
+     * @var \Drupal\menu_link_content\Entity\MenuLinkContent $menu_link_content
+     */
+    $menu_link_content = MenuLinkContent::load($id);
+    if ($menu_link_content !== NULL) {
+      // phpcs:disable
+      /* @phpstan-ignore-next-line */
+      $menu_link_content->link->title = 'Directory';
```
Comment (danahertzberg) [inline modules/custom/az_global_footer/az_global_footer.install:137]:
```suggestion
      $menu_link_content->link->title = 'Directory';
```
I'm unsure if this aspect will work. I'm looking to update the URL and the link title.

File: modules/custom/az_global_footer/az_global_footer.install [hunk 2]
```diff
@@ -77,3 +77,71 @@ function az_global_footer_update_1020701() {
 
   return t('Updated %count Campus Safety menu link(s) in the global footer.', ['%count' => $updated_count]);
 }
+
+/**
+ * Update Phonebook links
+ *
+ * Phonebook links in the global footer will be 
+ * updated with new primary domain.
+ */
```
Comment (joeparsons) [inline modules/custom/az_global_footer/az_global_footer.install:86]:
```suggestion
/**
 * Update Phonebook footer links.
 *
 * Phonebook links in the global footer will be updated with new primary domain.
 */
```

File: modules/custom/az_global_footer/az_global_footer.install [hunk 3]
```diff
@@ -77,3 +77,71 @@ function az_global_footer_update_1020701() {
...
+    ->accessCheck(FALSE)
+    ->condition('menu_name', 'az-footer-resources')
+    ->condition('title', 'Phonebook')
+    ->condition('link__uri', 'https://directory.arizona.edu/phonebook')
+    ->execute();
+
+  foreach ($menu_link_content_ids as $id) {
+    /**
+     * @var \Drupal\menu_link_content\Entity\MenuLinkContent $menu_link_content
+     */
```
Comment (joeparsons) [inline modules/custom/az_global_footer/az_global_footer.install:99]:
```suggestion
    /** @var \Drupal\menu_link_content\Entity\MenuLinkContent $menu_link_content */
```

File: modules/custom/az_global_footer/az_global_footer.install [hunk 4]
```diff
@@ -77,3 +77,71 @@ function az_global_footer_update_1020701() {
...
+    ->accessCheck(FALSE)
+    ->condition('menu_name', 'az-footer-resources')
+    ->condition('title', 'A-Z Index')
+    ->condition('link__uri', 'https://directory.arizona.edu/index')
+    ->execute();
+
+  foreach ($menu_link_content_ids as $id) {
+    /**
+     * @var \Drupal\menu_link_content\Entity\MenuLinkContent $menu_link_content
+     */
```
Comment (joeparsons) [inline modules/custom/az_global_footer/az_global_footer.install:132]:
```suggestion
    /** @var \Drupal\menu_link_content\Entity\MenuLinkContent $menu_link_content */
```

File: modules/custom/az_global_footer/az_global_footer.install [hunk 5]
```diff
@@ -77,3 +77,71 @@ function az_global_footer_update_1020701() {
...
+
+  return t('Updated %count Phonebook menu link(s) in the global footer.', ['%count' => $updated_count]);
+}
+
+/**
+ * Update Directory links
+ *
+ * Directory links in the global footer will be 
+ * updated with new primary domain.
+ */
```
Comment (joeparsons) [inline modules/custom/az_global_footer/az_global_footer.install:119]:
```suggestion
/**
 * Update Directory footer links.
 *
 * Directory links in the global footer will be updated with new primary domain.
 */
```

File: modules/custom/az_global_footer/az_global_footer.install [hunk 6]
```diff
@@ -77,3 +77,71 @@ function az_global_footer_update_1020701() {
...
   return t('Updated %count Campus Safety menu link(s) in the global footer.', ['%count' => $updated_count]);
 }
+
+/**
+ * Update Phonebook links
+ *
+ * Phonebook links in the global footer will be 
+ * updated with new primary domain.
+ */
+function az_global_footer_update_1020706() {
```
Comment (joeparsons) [inline modules/custom/az_global_footer/az_global_footer.install:87]:
```suggestion
function az_global_footer_update_1020702() {
```

File: modules/custom/az_global_footer/az_global_footer.install [hunk 7]
```diff
@@ -77,3 +77,71 @@ function az_global_footer_update_1020701() {
...
+  return t('Updated %count Phonebook menu link(s) in the global footer.', ['%count' => $updated_count]);
+}
+
+/**
+ * Update Directory links
+ *
+ * Directory links in the global footer will be 
+ * updated with new primary domain.
+ */
+function az_global_footer_update_1020706() {
```
Comment (joeparsons) [inline modules/custom/az_global_footer/az_global_footer.install:120]:
```suggestion
function az_global_footer_update_1020703() {
```

File: modules/custom/az_global_footer/az_global_footer.install (hunk 1 above)
Comment (danahertzberg) [reply modules/custom/az_global_footer/az_global_footer.install:131]: This did not work. URLs were updated correctly, but not the link title.

File: modules/custom/az_global_footer/az_global_footer.install [hunk 8]
```diff
@@ -77,3 +77,65 @@ function az_global_footer_update_1020701() {
...
+    ->condition('link__uri', 'https://directory.arizona.edu/index')
+    ->execute();
+
+  foreach ($menu_link_content_ids as $id) {
+    /** @var \Drupal\menu_link_content\Entity\MenuLinkContent $menu_link_content */
+    $menu_link_content = MenuLinkContent::load($id);
+    if ($menu_link_content !== NULL) {
+      // phpcs:disable
+      /* @phpstan-ignore-next-line */
+      $menu_link_content->link->title = 'Directory';
```
Comment (danahertzberg) [inline modules/custom/az_global_footer/az_global_footer.install:131]:
```suggestion
      $menu_link_content->title = 'Directory';
```

**Comment by danahertzberg** [comment]
Ready to merge!

**Review by trackleft — DISMISSED** [review]

File: modules/custom/az_global_footer/az_global_footer.install [hunk 9]
```diff
@@ -77,3 +77,65 @@ function az_global_footer_update_1020701() {
...
+    }
+  }
+
+  return t('Updated %count Phonebook menu link(s) in the global footer.', ['%count' => $updated_count]);
+}
+
+/**
+ * Update Directory footer links.
+ *
+ * Directory links in the global footer will be updated with new primary domain.
```
Comment (trackleft) [inline modules/custom/az_global_footer/az_global_footer.install:114]:
```suggestion
 * A-Z Index links in the global footer will be updated link to the campus directory instead.
```

File: modules/custom/az_global_footer/az_global_footer.install [hunk 10]
```diff
@@ -77,3 +77,65 @@ function az_global_footer_update_1020701() {
...
+      $menu_link_content->title = 'Directory';
+      /* @phpstan-ignore-next-line */
+      $menu_link_content->link->uri = 'https://directory.arizona.edu';
+      // phpcs:enable
+      $menu_link_content->save();
+      $updated_count++;
+    }
+  }
+
+  return t('Updated %count Directory menu link(s) in the global footer.', ['%count' => $updated_count]);
```
Comment (trackleft) [inline modules/custom/az_global_footer/az_global_footer.install:140]:
```suggestion
  return t('Updated %count directory menu link(s) in the global footer.', ['%count' => $updated_count]);
```

**Review by trackleft — APPROVED** [review]

**Review by joeparsons — APPROVED** [review]

**Pull request merged**